File: assign cookies.py
class Solution:
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        g.sort()
        s.sort()
        
        ans = 0
        j = 0
        
        for i in range(len(g)):
            while(j < len(s) and s[j] < g[i]):
                j += 1
            if(j< len(s)):
                ans += 1
                j += 1
        return ans

File: assign cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        std::sort(g.begin(), g.end()); // O(glog(g))
        std::sort(s.begin(), s.end()); // O(slog(s))
        
        int ans = 0;
        int j = 0;
        
        for(int r : g) { // O(g) + O(m)
            while(j<s.size() && s[j] < r) {
                ++j;
            }
            if(j<s.size()) {
                ++ans;
                ++j;
            }
        }
        return ans;
    }
};
